caffe2/operators/segment_reduction_op.h

```diff
@@ -1731,8 +1731,8 @@ class AbstractLengthsWithMainInputGradientOp : public Operator<Context> {
   }
 
   // Input layout:
-  //   orig_arg1, orig_arg2, ..., orig_argN, DATA_INPUT, SEGMENT_GRADS,
-  //      SEGMENT_LEGNTHS, [INDICES]
+  //   orig_arg1, orig_arg2, ..., orig_argN, SEGMENT_GRADS, LENGTHS,
+  //      DATA_INPUT, [INDICES]
   // orig_argXs represent original op's inputs and will be passed to the reducer
   // directly
   static constexpr int kNumInputs = ReducerGradient::originalInputs().size() +
```

Comment:
Looks like if SparseFused and GradientNeedIndices are both set, there are actually 5 extra inputs?

Reply:
Strangely, GradientNeedIndices is not used anywhere as far as I can tell. Let me merge this for now and look at this further.